tool/python/README.md

```diff
@@ -0,0 +1,330 @@
+
+
+## MLP Example
+
+An example (to generate job.conf for mnist)
+
+```
+X_train, X_test, workspace = mnist.load_data()
+
+m = Sequential('mlp', sys.argv)  
+
+m.add(Dense(2500, init='uniform', activation='tanh'))
+m.add(Dense(2000, init='uniform', activation='tanh'))
+m.add(Dense(1500, init='uniform', activation='tanh'))
+m.add(Dense(1000, init='uniform', activation='tanh'))
+m.add(Dense(500,  init='uniform', activation='tanh'))
+m.add(Dense(10, init='uniform', activation='softmax')) 
+
+sgd = SGD(lr=0.001, lr_type='step')
+topo = Cluster(workspace)
+m.compile(loss='categorical_crossentropy', optimizer=sgd, cluster=topo)
+m.fit(X_train, nb_epoch=1000, with_test=True)
+result = m.evaluate(X_test, batch_size=100, test_steps=10, test_freq=60)
+```
+
+## CNN Example
+
+An example (to generate job.conf for cifar10)
+
+```
+X_train, X_test, workspace = cifar10.load_data()
+
+m = Sequential('cnn', sys.argv)
+
+m.add(Convolution2D(32, 5, 1, 2, w_std=0.0001, b_lr=2))
+m.add(MaxPooling2D(pool_size=(3,3), stride=2))
+m.add(Activation('relu'))
+m.add(LRN2D(3, alpha=0.00005, beta=0.75))
+
+m.add(Convolution2D(32, 5, 1, 2, b_lr=2))
+m.add(Activation('relu'))
+m.add(AvgPooling2D(pool_size=(3,3), stride=2))
+m.add(LRN2D(3, alpha=0.00005, beta=0.75))
+
+m.add(Convolution2D(64, 5, 1, 2))
+m.add(Activation('relu'))
+m.add(AvgPooling2D(pool_size=(3,3), stride=2))
+
+m.add(Dense(10, w_wd=250, b_lr=2, b_wd=0, activation='softmax'))
+
+sgd = SGD(decay=0.004, lr_type='fixed', step=(0,60000,65000), step_lr=(0.001,0.0001,0.00001))
+topo = Cluster(workspace)
+m.compile(updater=sgd, cluster=topo)
+m.fit(X_train, nb_epoch=1000, with_test=True)
+result = m.evaluate(X_test, 1000, test_steps=30, test_freq=300)
+```
+
+
+## RBM Example
+```
+rbmid = 3                                                                                           
+X_train, X_test, workspace = rbm.load_data(rbmid)                                               
+                                                                                              
+m = Energy('rbm'+str(rbmid), sys.argv)
+
+out_dim = [1000, 500, 250]
+m.add(RBM(out_dim, w_std=0.1, b_wd=0)) 
+                                                                                         
+sgd = SGD(lr=0.1, decay=0.0002, momentum=0.8)                                
+topo = Cluster(workspace)                                                                    
+m.compile(optimizer=sgd, cluster=topo)                                                    
+m.fit(X_train, alg='cd', nb_epoch=6000)                            
+```
+
+## AutoEncoder Example
+```
+rbmid = 4
+X_train, X_test, workspace = ae.load_data(rbmid)
+
+m = Sequential('autoencoder')
+
+out_dim = [1000, 500, 250, 30]
+
+for i in range(1, rbmid):
+  m.add(Encoder(i, out_dim[i-1], activation='sigmoid'))
+m.add(Encoder(rbmid, out_dim[-1]))
+
+for i in range(rbmid, 1, -1):
+  m.add(Decoder(i, out_dim[i-2], activation='sigmoid'))
+m.add(Decoder(1, 784, activation='sigmoid'))
+
```

Comment:
keras may not provide methods for configuring parameter sharing.
if we want to share parameters between encoders and decoders, we may do it implicitly or explicitly as
1. m.set_param_share(true)
2. m.share(encoder[i], decoder[j])